lib/reek/code_comment.rb

```diff
@@ -11,31 +11,26 @@ class CodeComment
     CONFIG_REGEX = /:reek:(\w+)(:\s*\{.*?\})?/
 
     def initialize(text)
+      @config = Hash.new { |hash, key| hash[key] = {} }
       @text = text.gsub(CONFIG_REGEX) do
-        add_to_config($1, $2)
+        @config.merge! add_to_config($1, $2)
         ''
       end.gsub(/#/, '').gsub(/\n/, '').strip
     end
 
-    def config
-      @config ||= Hash.new { |hash, key| hash[key] = {} }
-    end
+    attr_reader :config
 
     def descriptive?
       text.split(/\s+/).length >= 2
     end
 
-    protected
+    private
 
     def add_to_config(smell, options)
       options ||= ': { enabled: false }'
-      config.merge! YAML.load(smell.gsub(/(?:^|_)(.)/) { $1.upcase } + options)
-      # TODO: extend this to all configs -------------------^
-      # TODO: extend to allow configuration of whole smell class, not just subclass
+      YAML.load(smell.gsub(/(?:^|_)(.)/) { $1.upcase } + options)
```

Comment:
Since we're already on it, could we maybe also refactor this mega-ugly line of code?